From b57427b67895edd12a120e1566b2e251a866bdf9 Mon Sep 17 00:00:00 2001
From: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@linux.intel.com>
Date: Mon, 21 May 2018 14:47:51 -0700
Subject: [PATCH 1463/2328] Revert "usb: phy-intel-dualrole: allow switching
 roles after gadget/host unregistered"

This reverts commit 143b50d27dabd33b9c65c6cda15836ed72bda1a8.

Signed-off-by: Kuppuswamy Sathyanarayanan <sathyanarayanan.kuppuswamy@linux.intel.com>
---
 drivers/usb/phy/phy-intel-dualrole.c | 20 +++++++++++++++-----
 1 file changed, 15 insertions(+), 5 deletions(-)

diff --git a/drivers/usb/phy/phy-intel-dualrole.c b/drivers/usb/phy/phy-intel-dualrole.c
index 0eaaf8e..a49bd1e 100644
--- a/drivers/usb/phy/phy-intel-dualrole.c
+++ b/drivers/usb/phy/phy-intel-dualrole.c
@@ -107,9 +107,14 @@ static int intel_dr_phy_set_host(struct usb_otg *otg, struct usb_bus *host)
 {
 	struct intel_dr_phy *intel_phy;
 
-	if (!otg || !host)
+	if (!otg)
 		return -ENODEV;
 
+	if (!host) {
+		otg->host = NULL;
+		return -ENODEV;
+	}
+
 	intel_phy = container_of(otg->usb_phy, struct intel_dr_phy, phy);
 
 	otg->host = host;
@@ -128,9 +133,14 @@ static int intel_dr_phy_set_peripheral(struct usb_otg *otg,
 {
 	struct intel_dr_phy *intel_phy;
 
-	if (!otg || !gadget)
+	if (!otg)
 		return -ENODEV;
 
+	if (!gadget) {
+		otg->gadget = NULL;
+		return -ENODEV;
+	}
+
 	intel_phy = container_of(otg->usb_phy, struct intel_dr_phy, phy);
 
 	otg->gadget = gadget;
@@ -185,9 +195,9 @@ static int intel_dr_phy_handle_notification(struct notifier_block *nb,
 	}
 
 	/*
-	 * Don't kick the state machine if host or device controller
-	 * were never registered before. Just wait to kick it when
-	 * set_host or set_peripheral.
+	 * don't kick the state machine if host or device controller
+	 * is not registered. Just wait to kick it when set_host or
+	 * set_peripheral.
 	 * */
 
 	if (host && gadget)
-- 
2.7.4

